File: src/test/java/com/jacuum/web/AlgosEndpointTest.java
package com.jacuum.web;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.web.servlet.MockMvc;
import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;

@SpringBootTest
@AutoConfigureMockMvc
class AlgosEndpointTest {

    private final MockMvc mvc;

    @Autowired
    AlgosEndpointTest(final MockMvc mvc) {
        this.mvc = mvc;
    }

    @Test void algosEndpointReturnsList() throws Exception {
        mvc.perform(get("/api/algos"))
            .andExpect(status().isOk())
            .andExpect(jsonPath("$").isArray())
            .andExpect(jsonPath("$.length()").value(org.hamcrest.Matchers.greaterThan(0)));
    }

    @Test void avatarsEndpointReturnsList() throws Exception {
        mvc.perform(get("/api/avatars"))
            .andExpect(status().isOk())
            .andExpect(jsonPath("$").isArray());
    }
}

File: src/test/kotlin/com/jacuum/web/AlgosEndpointTest.kt
package com.jacuum.web

import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.web.servlet.MockMvc
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get
import org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath
import org.springframework.test.web.servlet.result.MockMvcResultMatchers.status
import org.hamcrest.Matchers.greaterThan

@SpringBootTest
@AutoConfigureMockMvc
class AlgosEndpointTest @Autowired constructor(private val mvc: MockMvc) {

    @Test fun algosEndpointReturnsList() {
        mvc.perform(get("/api/algos"))
            .andExpect(status().isOk())
            .andExpect(jsonPath("$").isArray())
            .andExpect(jsonPath("$.length()").value(greaterThan(0)))
    }

    @Test fun avatarsEndpointReturnsList() {
        mvc.perform(get("/api/avatars"))
            .andExpect(status().isOk())
            .andExpect(jsonPath("$").isArray())
    }
}
